clippy_lints/src/derive.rs

```diff
@@ -293,7 +293,7 @@ fn check_ord_partial_ord<'tcx>(
 
 /// Implementation of the `EXPL_IMPL_CLONE_ON_COPY` lint.
 fn check_copy_clone<'tcx>(cx: &LateContext<'tcx>, item: &Item<'_>, trait_ref: &TraitRef<'_>, ty: Ty<'tcx>) {
-    if match_path(&trait_ref.path, &paths::CLONE_TRAIT) {
+    if cx.tcx.lang_items().clone_trait() == trait_ref.trait_def_id() {
```

Comment:
Maybe comparing `def_id`s isn't equivalent to the `match_path` method?

Reply:
The lang item might not be defined at the point. I didn't think that would actually happen though. I'll have a pr in a bit.